services/wallet/api.go

```diff
@@ -214,24 +214,21 @@ func (api *API) DeleteCustomTokenByChainID(ctx context.Context, chainID uint64,
 
 func (api *API) GetSavedAddresses(ctx context.Context) ([]SavedAddress, error) {
 	log.Debug("call to get saved addresses")
-	rst, err := api.s.savedAddressesManager.GetSavedAddressesForChainID(api.s.rpcClient.UpstreamChainID)
+	rst, err := api.s.savedAddressesManager.GetSavedAddresses()
 	log.Debug("result from database for saved addresses", "len", len(rst))
 	return rst, err
 }
 
 func (api *API) AddSavedAddress(ctx context.Context, sa SavedAddress) error {
 	log.Debug("call to create or edit saved address")
-	if sa.ChainID == 0 {
-		sa.ChainID = api.s.rpcClient.UpstreamChainID
-	}
 	_, err := api.s.savedAddressesManager.UpdateMetadataAndUpsertSavedAddress(sa)
 	log.Debug("result from database for create or edit saved address", "err", err)
 	return err
 }
 
-func (api *API) DeleteSavedAddress(ctx context.Context, address common.Address) error {
+func (api *API) DeleteSavedAddress(ctx context.Context, address common.Address, ens string, isTest bool) error {
```

Comment:
isTest can be inferred right?

Reply:
maybe not in the API but savedAddressesManager should have access to the multiaccounts